src/_pytest/capture.py

```diff
@@ -695,14 +710,65 @@ def __init__(self, fd, tmpfile=None):
         self.tmpfile = tmpfile
 
 
+class OrderedCapture(SysCapture):
+    """Capture class that keeps streams in order."""
+
+    streams = collections.deque()  # type: collections.deque
```

Comment:
seems a bit odd for this to be a class member and not an instance member (as written this makes it effectively a global variable).  I'd rather see it as an instance variable

Reply:
Definitely, but see my comment in in https://github.com/pytest-dev/pytest/pull/6690#issuecomment-629804771:

> As @bluetech commented, it would probably better if we could get away with the global variables and tight coupling between OrderedCapture and OrderedWriter, but looking at the overall design, I see why it was coded this way: currently MultiCapture always demands 3 captures (stdout, stderr, stdin), so we need a way to communicate/order the writes.

IOW I don't know how to easily fix this, as @aklajnert also commented.

Reply:
I think all that's needed is to grow another api on the capture classes which represents combined? if that's possible I think that would also fix the problem below of the backward-class-coupling

Reply:
Hmm but currently who deals with each capture class is `MultiCapture`, which always creates 3 capture instances (`Capture(0)`, `Capture(1)`, and `Capture(2)`).

Or do you mean `MultiCapture` to grow this new API, and forward to the underlying `Capture` classes?

Reply:
oh I see :/ maybe nevermind then

Reply:
Yeah it is a bit frustrating, that's why I think we have to rethink the whole design (in a future PR that is).